### PR TITLE
Product Refactoring + unit tests for controller and service class

### PR DESCRIPTION
**Pull request by TatjanaPro**

### CODE REVIEW

File: auction-app/src/main/java/com/auctionapp/model/product/Product.java [hunk 1]
```diff
@@ -24,16 +24,11 @@ public class Product {
     @Basic
     @Column(name = "name")
     @NotBlank(message = "Name is mandatory")
-    @Size(min = 4, max = 50, message = "Name must be between 4 and 50 characters")
+    @Size(min = 3, max = 50, message = "Name must be between 4 and 50 characters")
```
Comment (VaivaSvegzdaite) [inline auction-app/src/main/java/com/auctionapp/model/product/Product.java]: 4->3 in message probably should change too?

File: auction-app/src/main/java/com/auctionapp/service/ProductService.java [hunk 2]
```diff
@@ -1,23 +1,41 @@
...
         this.productRepository = productRepository;
+        this.userService = userService;
     }
 
-    public void createProduct(Product product) {
+    public ResponseEntity<String> createProduct(Product product) {
+        if (product.getId() != null && product.getId() != 0) {
+            return ResponseEntity.badRequest().body("Invalid product ID");
+        }
+        //productRepository.save(product);
```
Comment (VaivaSvegzdaite) [inline auction-app/src/main/java/com/auctionapp/service/ProductService.java]: as save is lower, probably this commented code is not needed

**Review by VaivaSvegzdaite — COMMENTED** [review]
Couple notes in code

File: auction-app/src/main/java/com/auctionapp/model/product/Product.java (hunk 1 above)
Comment (TatjanaPro) [reply auction-app/src/main/java/com/auctionapp/model/product/Product.java]: Changed

File: auction-app/src/main/java/com/auctionapp/service/ProductService.java (hunk 2 above)
Comment (TatjanaPro) [reply auction-app/src/main/java/com/auctionapp/service/ProductService.java]: Deleted

**Review by VaivaSvegzdaite — APPROVED** [review]
Thanks

**Pull request merged**